src/scraper/__main__.py

```diff
@@ -1,35 +1,81 @@
+        self.judicial_officers = judicial_officers #"Judicial Officers to scrape. For example, -j 'mr. something' 'Rob, Albert'. By default, it will scrape all JOs.",
+        self.no_overwrite = no_overwrite #"Switch to don't overwrite cached html files if you want to speed up the process (but may not get the most up to date version).",
+        self.log = log #"Set the level to log at.",
+        self.court_calendar_link_text = court_calendar_link_text #"This is the link to the Court Calendar search page at default.aspx, usually it will be 'Court Calendar', but some sites have multiple calendars e.g. Williamson",
+        self.test = test #"If this parameter is present, the script will stop after the first case is scraped.",
+        self.case_number = case_number #"If a case number is entered, only that single case is scraped. ex. 12-2521CR",
+        self.location = location # ??
+
+        # These are class scraper-level fields that need to be stored and used from function to function. 
+        self.odyssey_version = None
+        self.base_url = None
+        self.main_soup = None
+        self.main_page_html = None
+        self.search_url = None
+        self.search_page_html = None
+        self.search_soup = None
+        self.hidden_values = None
+        self.judicial_officer_to_ID = None
+        self.START_TIME = None
+
         # allow bad ssl and turn off warnings
+        self.session = requests.Session()
         self.session.verify = False
         requests.packages.urllib3.disable_warnings(
             requests.packages.urllib3.exceptions.InsecureRequestWarning
         )
+
+        # configure the logger
         self.logger = logging.getLogger(name="pid: " + str(os.getpid()))
         logging.basicConfig()
-        logging.root.setLevel(level=args.log)
+        logging.root.setLevel(level=self.log)
 
         # make cache directories if not present
-        self.case_html_path = os.path.join(
-            os.path.dirname(__file__), "..", "..", "data", self.county, "case_html"
-        )
-        os.makedirs(self.case_html_path, exist_ok=True)
+        if not self.test:
+            self.case_html_path = os.path.join(
+                os.path.dirname(__file__), "..", "..", "data", self.county, "case_html"
+            )
+            os.makedirs(self.case_html_path, exist_ok=True)
+            # Makes a list of existing cases on the device
+            self.cached_case_list = [file_name.split(".")[0] for file_name in os.listdir(self.case_html_path)]
+        else: # if test = True
+            self.case_html_path = os.path.join(
+                os.path.dirname(__file__), "..", "..", "resources", 'test_files', 'test_data', self.county, "case_html"
+            )
+            os.makedirs(self.case_html_path, exist_ok=True)
+            # Makes a list of existing cases on the device
+            self.cached_case_list = [file_name.split(".")[0] for file_name in os.listdir(self.case_html_path)]
```

Comment:
📓  Just noting here as we discussed in-person, this is probably an ideal place to use Dependency Injection.